src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -1996,37 +2078,54 @@ public FormValidation doCheckCredentialsId(@CheckForNull @AncestorInPath Item co
         public FormValidation doValidateRepositoryUrlAndCredentials(@CheckForNull @AncestorInPath Item context,
                                                                     @QueryParameter String repositoryUrl,
                                                                     @QueryParameter String credentialsId) {
+
+            // Start by verifying we have a well-formed GitHub API URL
+            GitHubRepositoryInfo info;
+
+            try {
+                info = GitHubRepositoryInfo.forRepositoryUrl(repositoryUrl);
+            } catch (Throwable e) {
```

Comment:
No matter what error you get from this method call, report it to the UI as a form validation error. It will always look better that way that the default UI exception formatting.

Reply:
Also: `forRepositoryUrl` doesn't throw an exception so catching it gets your compiler problems.  I'll try changing it.  https://github.com/jenkinsci/github-branch-source-plugin/pull/254/files#diff-37978319a92a70efbd5003352734d2b5R81